cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnel.java

```diff
@@ -120,20 +122,16 @@ public synchronized void tearDownVpn() {
   /**
    * Connects a tunnel between a Shadowsocks proxy server and the VPN TUN interface.
    *
-   * @param host is  IP address of the SOCKS proxy server.
-   * @param port is the port of the SOCKS proxy server.
-   * @param password is the password of the Shadowsocks proxy.
-   * @param cipher is the encryption cipher used by the Shadowsocks proxy.
+   * @param client provides access to the Shadowsocks proxy.
    * @throws IllegalArgumentException if |socksServerAddress| is null.
    * @throws IllegalStateException if the VPN has not been established, or the tunnel is already
    *     connected.
    * @throws Exception when the tunnel fails to connect.
    */
-  public synchronized void connectTunnel(final String host, int port, final String password,
-      final String cipher, boolean isUdpEnabled) throws Exception {
+  public synchronized void connectTunnel(final shadowsocks.Client client, boolean isUdpEnabled) throws Exception {
```

Comment:
Note for future: we can probably get rid of the isUdpEnabled option by passing a Client that knows it needs to manipulate the DNS requests.

Reply:
Added a TODO